integrations/datadog/client.py

```diff
@@ -404,48 +415,65 @@ async def _fetch_metrics_for_services(
 
             yield metrics
 
-    async def get_metrics(
-        self,
-        metric_query: str,
-        env: str = "*",
-        service: str = "*",
-        time_window: int = FETCH_WINDOW_TIME_IN_MINUTES,
-    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+    async def get_metrics(self) -> AsyncGenerator[list[dict[str, Any]], None]:
         """
         Fetches metrics for specified services and environment.
 
         Args:
             metric_query (str): The Datadog metric to fetch (e.g., "avg:container.cpu.usage").
             env (str): The environment to filter by, or "*" to fetch metrics for all environments.
             service (str): The service ID to filter by, or "*" to fetch metrics for all services.
-            time_window (int): Time window in minutes for fetching metrics (default: FETCH_WINDOW_TIME_IN_MINUTES).
+            timeframe (int): Time window in minutes for fetching metrics (default: FETCH_WINDOW_TIME_IN_MINUTES).
 
         Yields:
             AsyncGenerator[list[dict[str, Any]], None]: Each individual metric as it's fetched.
         """
+        params: DatadogSelector = typing.cast(
+            DatadogResourceConfig, event.resource_config
+        ).selector.datadog_selector
+
+        logger.info(f"Fetching metrics for params: {params}")
+
+        env_tag, env_value = self.get_metric_config(params.env)
+        service_tag, service_value = self.get_metric_config(params.service)
+
+        logger.info(f"env_tag: {env_tag}, env_value: {env_value}")
+        logger.info(f"service_tag: {service_tag}, service_value: {service_value}")
```

Comment:
I think this should be inside the `main.py` and not in the client. as the client needs to be just get the variables and not be familiar with the event.resource_config.